### PR TITLE
Add plugin API callback documentation

### PR DESCRIPTION
**Pull request by brentmackinnon**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

 Create woocommerce-plugin-api-callback.md - migrated from wccom. 

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1.
2.
3.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: docs/extension-development/woocommerce-plugin-api-callback.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# WooCommerce Plugin API Callback Documentation
+
+## Overview
+
+This document provides a detailed guide on how to use the WooCommerce Plugin API to initiate callbacks for plugin actions, especially for gateways and classes not initialized by default.
```
Comment (lsinger) [inline docs/extension-development/woocommerce-plugin-api-callback.md]: It's really very barebones in my opinion, so let's at least remove the "detailed" claim here. 

```suggestion
This document provides a guide on how to use the WooCommerce Plugin API to initiate callbacks for plugin actions, especially for gateways and classes not initialized by default.
```

File: docs/extension-development/woocommerce-plugin-api-callback.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+## Callback URL Structure
+
+Before WooCommerce 2.0, use:
+
+```php
```
Comment (lsinger) [inline docs/extension-development/woocommerce-plugin-api-callback.md]: This is not PHP, it's a URL -- we should put it into single quotes like so:

`https://...`

Putting it onto its own line seems fine here.

File: docs/extension-development/woocommerce-plugin-api-callback.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+http://yoursite.com/?wc-api=CALLBACK
+```
+
+In WooCommerce 2.0 or later, use the endpoint:
+
+```php
```
Comment (lsinger) [inline docs/extension-development/woocommerce-plugin-api-callback.md]: See above -- this is a URL, not PHP code.

File: docs/extension-development/woocommerce-plugin-api-callback.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+## Overview
+
+This document provides a detailed guide on how to use the WooCommerce Plugin API to initiate callbacks for plugin actions, especially for gateways and classes not initialized by default.
+
+## Callback URL Structure
+
+Before WooCommerce 2.0, use:
+
+```php
+http://yoursite.com/?wc-api=CALLBACK
```
Comment (lsinger) [inline docs/extension-development/woocommerce-plugin-api-callback.md]: The `example.com` and `example.org` domains exist for specifically this purpose. Let's use them. Also, using HTTPS is a nicer default. 

```suggestion
https://example.com/?wc-api=CALLBACK
```

File: docs/extension-development/woocommerce-plugin-api-callback.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+```
+
+In WooCommerce 2.0 or later, use the endpoint:
+
+```php 
+http://yoursite.com/wc-api/CALLBACK/
```
Comment (lsinger) [inline docs/extension-development/woocommerce-plugin-api-callback.md]: See above. 

```suggestion
https://example.com/wc-api/CALLBACK/
```

**Review by lsinger — CHANGES_REQUESTED** [review]
Thanks for working on this! I did leave a few comments with changes we should make.

**Comment by brentmackinnon** [comment]
awesome thanks @lsinger! I've made those adjustments.

**Review by lsinger — APPROVED** [review]
Thank you for addressing my comments, @brentmackinnon. This looks good to go. 👍

**Comment by lsinger** [comment]
@brentmackinnon in the spirit of friendly, constructive feedback, it would be nice if in the future you could ...

- edit the default commit messages that GitHub provides ("Update FILENAME") to something more meaningful ("use example.com instead of yoursite.com" or "tweak the wording in the opening paragraph")
- do the same with the PR title (which will end up in the repo commit history) -- I've renamed it to "Add plugin API callback documentation" from "Create woocommerce-plugin-api-callback.md"
- use the "re-request review" button when a PR is ready for another round of reviews -- that way your PR will show up in [your reviewer's review queue](https://github.com/pulls/review-requested), which will make it less likely that they will miss it:

Screenshot 2023-11-24 at 09 59 48

I know I'm dropping a lot of stuff here but again, my intention is purely kind and helpful. ☺️

Thanks for driving us all to make our documentation better!

**Pull request merged**